cmd/wallet-web/src/pages/OIDCShare.vue

```diff
@@ -295,6 +294,27 @@ export default {
 
       this.sharing = false;
     },
+    async validateRelyingParty() {
+      const registrationUri = this.$route.query.registration_uri;
+      if (registrationUri) {
+        // Obtaining registration metadata according to the spec:
+        // https://openid.net/specs/openid-connect-self-issued-v2-1_0.html#name-relying-party-metadata-reso
+        const registrationResponse = await fetch(registrationUri);
+        const registrationMetadata = await registrationResponse.json();
+        // Use the registration metadata jwks_uri param to get the public key
+        // data from jwks_uri
+        // https://datatracker.ietf.org/doc/html/rfc7517s
+        const jwksUri = registrationMetadata.jwks_uri;
+        const jwksResponse = await fetch(jwksUri);
+        const jwksKeys = await jwksResponse.json();
+        // We will get an array of keys
+        // we may need to search for the specific public keys in the future
+        // taking the first one for now
+        this.rpPublicKey = jwksKeys.keys[0];
+        this.generateIdToken(this.rpPublicKey);
+        this.generateVPToken();
```

Comment:
Can we move generate functions outside of validate func ? Lets have single responsibility functions.